components/elm/src/biogeophys/SoilHydrologyMod.F90

```diff
@@ -387,14 +401,26 @@ subroutine Infiltration(bounds, num_hydrologyc, filter_hydrologyc, num_urbanc, f
                 qflx_evap(c)=qflx_ev_soil(c)
              endif
 
+             !0. partition grid-level floodplain inundation volume and fraction to each column
+             if (use_lnd_rof_two_way) then
+                if (mod(get_nstep()-1,6) == 1 .or. get_nstep() <= 1) then                 
+                  h2orof(c)      = atm2lnd_vars%h2orof_grc(g) * wtgcell(c) 
+                  frac_h2orof(c) = atm2lnd_vars%frac_h2orof_grc(g) * wtgcell(c)
+                endif
+                ! TODO: add inundfrac from ocean 
+                if ( frac_h2orof(c) > 1 - fsno - frac_h2osfc(c) ) then
```

Comment:
How about frac_h2orof(c) = min(frac_h2orof(c), 1 - fsno - frac_h2osfc(c))

Reply:
Thanks for the suggestion. I prefer to use the if statement, because it will be more flexible when we introduce ocean inundation into ELM in the future. For example, we need to address the condition that` frac_h2orof + frac_h2oocn + frac_h2osfc + fsno > 1`.